intellij-toml/core/src/main/kotlin/org/toml/ide/intentions/TomlExpandInlineTableIntention.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.toml.ide.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiParserFacade
+import com.intellij.psi.util.parentsOfType
+import com.intellij.refactoring.suggested.endOffset
+import org.toml.lang.psi.*
+
+class TomlExpandInlineTableIntention : TomlElementBaseIntentionAction<TomlExpandInlineTableIntention.Context>() {
+    override fun getText(): String = "Expand into separate table"
+    override fun getFamilyName(): String = text
+
+    override fun findApplicableContext(project: Project, editor: Editor, element: PsiElement): Context? {
+        val keyValue = element.parentsOfType<TomlKeyValue>().firstOrNull {
+            val hasInlineTableValue =  it.value is TomlInlineTable
+            val hasTableOrArrayParent = it.parent is TomlKeyValueOwner && it.parent is TomlHeaderOwner
+            val hasTomlFileParent = it.parent is TomlFile
+
+            hasInlineTableValue && (hasTableOrArrayParent || hasTomlFileParent)
+        } ?: return null
+
+        val inlineTable = keyValue.value as TomlInlineTable
+        val parentTable = keyValue.parent as? TomlKeyValueOwner
+        return Context(keyValue, inlineTable, parentTable)
+    }
+
+    override fun invoke(project: Project, editor: Editor, ctx: Context) {
+        val key = ctx.keyValue.key.text
+        val parentTable = ctx.parentTable
+
+        val newTableKey = if (parentTable != null) {
+            val parentHeader = (parentTable as? TomlHeaderOwner)?.header ?: return
+            val parentTableKey = parentHeader.key?.text ?: return
+            "$parentTableKey.$key"
+        } else {
+            key
+        }
+
+        val newTable = TomlPsiFactory(project).createTable(newTableKey)
+        val psiParserFacade = PsiParserFacade.SERVICE.getInstance(project)
+        for (entry in ctx.inlineTable.entries) {
+            newTable.add(psiParserFacade.createWhiteSpaceFromText("\n"))
```

Comment:
Let's make `createWhitespace`/`createNewline` in `TomlPsiFactory` just like in `RsPsiFactory`